### PR TITLE
SAK-40350 Calendar subscribing to itself can loop

### PR DESCRIPTION
**Pull request by maurercw**
Move the loop checking code above the getCalendarReferences() call since that would trigger more looping.
Add a null check to prevent NPE

### CODE REVIEW

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 1]
```diff
@@ -5737,6 +5729,16 @@ protected void handleAccessIcalCommon(HttpServletRequest req, HttpServletRespons
 			log.warn("Reject internal request for: "+ calRef);
 			return;
 		}
+
+		// Extract the alias name to use for the filename.
+		List alias =  m_aliasService.getAliases(calRef);
```
Comment (ern) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java]: add generics i.e. `List<Alias>`

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java [hunk 2]
```diff
@@ -6198,7 +6198,7 @@ public void doUpdate(RunData runData, Context context)
 				if(returnState.compareTo(CalendarAction.STATE_INITED) == 0) {
 					returnState = this.defaultStateView;
 				}
-				if (returnState.endsWith("!!!fromDescription"))
+				if (returnState != null && returnState.endsWith("!!!fromDescription"))
```
Comment (ern) [inline calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java]: maybe use 'StringUtils.endsWith(returnState, "!!!fromDescription")'

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java (hunk 1 above)
Comment (maurercw) [reply calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java]: I didn't "really" make this change, just moved the existing code around, but I can still do it.

File: calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java (hunk 2 above)
Comment (maurercw) [reply calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java]: Change made.

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java (hunk 1 above)
Comment (maurercw) [reply calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java]: Change made.

**Comment by jonespm** [comment]
Also checked that this fixes the issue and subscriptions from external icals still work.

**Pull request merged**